src/vaults.lua

```diff
@@ -74,10 +86,28 @@ function vaults.vaultedTransfer(from, recipient, qty, lockLengthMs, currentTimes
 		balance = qty,
 		startTimestamp = currentTimestamp,
 		endTimestamp = currentTimestamp + lockLengthMs,
+		controller = revokable and from or nil,
 	})
 	return newVault
 end
 
+--- Revokes a vaulted transfer
+---@param from WalletAddress
+---@param recipient WalletAddress
+---@param vaultId VaultId
+---@param currentTimestamp Timestamp
+---@return Vault
+function vaults.revokeVaultedTransfer(from, recipient, vaultId, currentTimestamp)
+	local vault = vaults.getVault(recipient, vaultId)
+	assert(vault, "Vault not found.")
+	assert(vault.controller == from, "Only the controller can revoke the vault.")
+	assert(currentTimestamp <= vault.endTimestamp, "Vault has ended.")
```

Comment:
I think technically if it were `=` then the prune before this is called would have run. I also don't think revoking at the time at which the vault now belongs to the recipient is fair. Consider `<` only.

Reply:
+1

Reply:
updated in this commit: [c8ed870](https://github.com/ar-io/ar-io-network-process/pull/334/commits/c8ed870db47188043ce463ff144ff6b97a18401a)